pandas/tests/io/json/test_pandas.py

```diff
@@ -985,12 +985,46 @@ def test_tz_range_is_utc(self):
         df = DataFrame({'DT': dti})
         assert dumps(df, iso_dates=True) == dfexp
 
-    def test_read_jsonl(self):
+    def test_read_inline_jsonl(self):
         # GH9180
         result = read_json('{"a": 1, "b": 2}\n{"b":2, "a" :1}\n', lines=True)
         expected = DataFrame([[1, 2], [1, 2]], columns=['a', 'b'])
         assert_frame_equal(result, expected)
 
+    @pytest.yield_fixture(scope="function")
+    def testbucket_conn(self):
+        """ Fixture for test_read_s3_jsonl"""
+        boto3 = pytest.importorskip('boto3')
+        moto.mock_s3().start()   # Start and stop mocking only once, surrounding the test run
+                                 # to ensure single context is kept.
+
+        conn = boto3.client('s3')
+        conn.create_bucket(Bucket='testbucket')
+        yield conn
+
+        moto.mock_s3().stop()
+
+    def test_read_s3_jsonl(self, testbucket_conn):
+        # GH17200
+        testbucket_conn.put_object(Body=b'{"a": 1, "b": 2}\n{"b":2, "a" :1}\n', Key='items.jsonl', Bucket='testbucket')
+
+        result = read_json('s3://testbucket/items.jsonl', lines=True)
+        expected = DataFrame([[1, 2], [1, 2]], columns=['a', 'b'])
+        assert_frame_equal(result, expected)
+
+    def test_read_local_jsonl(self):
+        # GH17200
+        fname = "./tmp_items.jsonl"
```

Comment:
Can you use the `tm.ensure_clean` context manager here? Then you don't have to worry about the `try / finally`.